pkg/tuned/cmd/render/render.go

```diff
@@ -126,18 +129,38 @@ func render(inputDir []string, outputDir string) error {
 		}
 	}
 
-	perfProfilesTuneDProfiles := make(map[*performancev2.PerformanceProfile]*tunedv1.Tuned, len(perfProfiles))
-	for _, profile := range perfProfiles {
-		tunedFromPP, err := tuned.NewNodePerformance(profile)
-		if err != nil {
-			klog.Errorf("Unable to get tuned. error : %v", err)
-			return fmt.Errorf("unable to get tuned from profile: %w", err)
-		}
-		perfProfilesTuneDProfiles[profile] = tunedFromPP
-		// add tuned from PP to the list
-		tuneD = append(tuneD, tunedFromPP)
+	mcp := findMachineConfigPoolByName(mcPools, mcpName)
+	if mcp == nil {
+		klog.Errorf("Unable to find MachineConfigPool:%q in input folders", mcpName)
+		return fmt.Errorf("Unable to find MachineConfigPool:%q in input folders", mcpName)
+	}
+
+	filteredPerformanceProfiles, err := filterPerformanceProfilesByMachineConfigPool(perfProfiles, mcp)
+	if err != nil {
+		klog.Errorf("Unable to find a PerformanceProfile that matches the MachineConfigPool %s. error : %v", mcpName, err)
+		return fmt.Errorf("Unable to get PerformanceProfile to apply using MachineConfigPool %s. error : %w", mcpName, err)
 	}
 
+	if len(filteredPerformanceProfiles) == 0 {
+		klog.Warningf("Cannot find any PerformanceProfile applicable to MachineConfigPool %s.", mcpName)
+		return fmt.Errorf("Cannot find any PerformanceProfile complying with MachineConfigPool %s.", mcpName)
+	}
+
+	if len(filteredPerformanceProfiles) > 1 {
+		klog.Warningf("Found %d PerformanceProfiles for MachineConfigPool %s, only ONE is allowed.", len(filteredPerformanceProfiles), mcpName)
```

Comment:
Tested it with one performance profile, and we are exiting here - deeds debugging

Reply:
yep, problem found and solved.